### PR TITLE
refactor: to header only

### PR DESCRIPTION
**Pull request by justforlxz**

### CODE REVIEW

File: src/types/qwkeyboardshortcutsinhibitv1.h [hunk 1]
```diff
@@ -3,56 +3,30 @@
...
+extern "C" {
+#include <wlr/types/wlr_keyboard_shortcuts_inhibit_v1.h>
+}
 
 QW_BEGIN_NAMESPACE
 
-class QWDisplay;
-class QWKeyboardShortcutsInhibitorV1Private;
-class QW_EXPORT QWKeyboardShortcutsInhibitorV1 : public QWWrapObject
+class QW_CLASS_REINTERPRET_CAST(keyboard_shortcuts_inhibitor_v1)
```
Comment (zccrs) [inline src/types/qwkeyboardshortcutsinhibitv1.h:14]: qw_class_object

File: src/types/qwlayershellv1.h [hunk 2]
```diff
@@ -3,75 +3,50 @@
...
 
 QW_BEGIN_NAMESPACE
 
-class QWDisplay;
-class QWSurface;
-class QWLayerSurfaceV1;
-class QWLayerShellV1Private;
-class QWXdgPopup;
-class QW_EXPORT QWLayerShellV1 : public QWWrapObject
+class qw_xdg_popup;
```
Comment (zccrs) [inline src/types/qwlayershellv1.h:21]: remove

File: src/types/qwlayershellv1.h [hunk 3]
```diff
@@ -3,75 +3,50 @@
...
+extern "C" {
+#include <math.h>
+#define namespace scope
+#define static
+#include <wlr/types/wlr_layer_shell_v1.h>
+#undef namespace
+#undef static
+}
 
 using wlr_surface_iterator_func_t = void (*)(wlr_surface *surface, int sx, int sy, void *data);
```
Comment (zccrs) [inline src/types/qwlayershellv1.h:17]: remove

File: src/types/qwlayershellv1.h [hunk 4]
```diff
@@ -3,75 +3,50 @@
...
-public:
-    static QWLayerShellV1 *create(QWDisplay *display, uint32_t version);
-    inline wlr_layer_shell_v1 *handle() const {
-        return QWObject::handle<wlr_layer_shell_v1>();
-    }
 
-    static QWLayerShellV1 *get(wlr_layer_shell_v1 *handle);
-    static QWLayerShellV1 *from(wlr_layer_shell_v1 *handle);
+public:
+    QW_SIGNAL(new_popup, qw_xdg_popup*)
```
Comment (zccrs) [inline src/types/qwlayershellv1.h:29]: wlr_xdg_popup

File: src/types/qwoutput.h [hunk 5]
```diff
@@ -3,147 +3,103 @@
...
+#include <qwobject.h>
+
+extern "C" {
+#include <math.h>
+#define static
+#include <wlr/types/wlr_output.h>
+#undef static
+}
+
 struct wlr_output_event_damage;
```
Comment (zccrs) [inline src/types/qwoutput.h:15]: remove

File: src/types/qwoutput.h [hunk 6]
```diff
@@ -3,147 +3,103 @@
...
 #if WLR_VERSION_MINOR < 18
-    void setMode(wlr_output_mode *mode);
-    void setCustomMode(const QSize &size, int32_t refresh);
-    void setTransform(wl_output_transform_t wl_output_transform);
-    void enableAdaptiveSync(bool enabled);
-    void setRenderFormat(uint32_t format);
-    void setScale(float scale);
-    void setSubpixel(wl_output_subpixel_t wl_output_subpixel);
-    void setDamage(pixman_region32 *damage);
+    QW_FUNC_MEMBER(output, set_mode)
```
Comment (zccrs) [inline src/types/qwoutput.h:39]: Have destroy function?

File: src/types/qwoutputlayout.h [hunk 7]
```diff
@@ -3,63 +3,49 @@
...
-struct wlr_output_layout_output;
+extern "C" {
+#include <math.h>
+#define static
+#include <wlr/types/wlr_output_layout.h>
+#undef static
+#include <wlr/util/box.h>
+}
 
 typedef uint32_t wlr_direction_t;
```
Comment (zccrs) [inline src/types/qwoutputlayout.h:16]: remove

File: src/types/qwoutputlayout.h [hunk 8]
```diff
@@ -3,63 +3,49 @@
...
 
 QW_BEGIN_NAMESPACE
 
-class QWDisplay;
-class QWOutput;
-class QWRenderer;
-class QWOutputLayoutPrivate;
-class QW_EXPORT QWOutputLayout : public QWWrapObject
+class QW_CLASS_REINTERPRET_CAST(output_layout_output)
 {
```
Comment (zccrs) [inline src/types/qwoutputlayout.h:21]: Why is empty?

**Review by zccrs — CHANGES_REQUESTED** [review]

**Review by zccrs — CHANGES_REQUESTED** [review]

File: src/types/qwlayershellv1.h [hunk 9]
```diff
@@ -3,75 +3,47 @@
...
+    QW_SIGNAL(new_popup, xdg_popup*)
 
-private:
-    QWLayerShellV1(wlr_layer_shell_v1 *handle, bool isOwner);
-    ~QWLayerShellV1() = default;
+public:
+    QW_FUNC_STATIC(layer_surface_v1, create)
+    QW_FUNC_STATIC(layer_surface_v1, try_from_wlr_surface)
+
+    QW_FUNC_MEMBER(layer_surface_v1, destroy)
```
Comment (zccrs) [inline src/types/qwlayershellv1.h:31]: Is it has the wlr_layer_surface_v1_destroy function?

**Review by zccrs — CHANGES_REQUESTED** [review]

File: src/types/qwlayershellv1.h (hunk 9 above)
Comment (justforlxz) [reply src/types/qwlayershellv1.h:31]: `void wlr_layer_surface_v1_destroy(struct wlr_layer_surface_v1 *surface);`

**Pull request merged**